doc/source/development/rfc/rfc104_gdal_cli.rst

```diff
@@ -0,0 +1,1175 @@
+          "description":"Output vector dataset",
+          "min_count":1,
+          "max_count":1,
+          "category":"Base",
+          "dataset_type":[
+            "vector"
+          ],
+          "input_flags":[
+            "name",
+            "dataset"
+          ],
+          "output_flags":[
+            "dataset"
+          ]
+        }
+      ]
+    }
+
+.. _rfc104_gdal_vector_pipeline_examples:
+
+A few invocations of "gdal vector pipeline"
++++++++++++++++++++++++++++++++++++++++++++
+
+  .. code-block:: shell
+
+     # The use of the '!' as a step separator is to prevent Unix or Windows shells from
+     # trying to use other processes for the "reproject" or "write" steps.
+     # Below is a single-process pipeline.
+     $ gdal vector pipeline read poly.gpkg ! reproject --dst-crs=EPSG:4326 ! write out.parquet --overwrite
+
+     # Alternative without the "vector" and "pipeline" subcommands, and with --progress
+     $ gdal read poly.gpkg ! reproject --dst-crs=EPSG:4326 ! write out.parquet --overwrite  --progress
+
+     # Alternative using an explicit --pipeline switch, and given the quoting, we can use the '|' character
+     $ gdal vector pipeline --pipeline="read poly.gpkg | reproject --dst-crs=EPSG:4326 | write out.parquet --overwrite"
+
+     # Works also as a quoted positional argument, and without the "vector" subcommand
+     $ gdal pipeline --progress "read poly.gpkg | reproject --dst-crs=EPSG:4326 | write out.parquet --overwrite"
+
+
+Detailed usage help message of "gdal vector pipeline"
++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+  .. code-block:: shell
+
+    $ gdal vector pipeline --help
+    Usage: gdal vector pipeline [OPTIONS] <PIPELINE>
+
+    Process a vector dataset.
+
+    Positional arguments:
+
+    Common Options:
+      -h, --help    Display help message and exit
+      --json-usage  Display usage as JSON document and exit
+      --progress    Display progress bar
+
+    <PIPELINE> is of the form: read [READ-OPTIONS] ( ! <STEP-NAME> [STEP-OPTIONS] )* ! write [WRITE-OPTIONS]
+
+    Example: 'gdal vector pipeline --progress ! read in.gpkg ! \
+                   reproject --dst-crs=EPSG:32632 ! write out.gpkg --overwrite'
+
+    Potential steps are:
+
+    * read [OPTIONS] <INPUT>
+    ------------------------
+
+    Read a vector dataset.
+
+    Positional arguments:
+      -i, --input <INPUT>                                  Input vector dataset [required]
+
+    Options:
+      -l, --layer, --input-layer <INPUT-LAYER>             Input layer name(s) [may be repeated]
+
+    Advanced Options:
+      --if, --input-format <INPUT-FORMAT>                  Input formats [may be repeated]
+      --oo, --open-option <KEY=VALUE>                      Open options [may be repeated]
+
+    * filter [OPTIONS]
+    ------------------
+
+    Filter.
+
+    Options:
+      --bbox <BBOX>                                        Bounding box as xmin,ymin,xmax,ymax [may be repeated]
```

Comment:
What does it mean `may be repeated`? The union or the intersection of the bboxes? Or anything else?

Reply:
You mentioned somewhere else that `argparse` is not going to be used, right? (it is a pity)

For me a repeating option (as it says "may" be repeated) is when I can repeat the option a variable number of times. Like `--co`. Here I would just remove `[may be repeated]`. BBOX is explained that is a 4-tuple.

Reply:
Maybe some day we want to have more dimensions into bbox but I guess that it is not so hard to add support for 3D/4D bbox later if needed.

Reply:
bbox now declared as SetRepeatedArgAllowed(false). New usage no longer mentions [may be repeated]